practices/oocss.md

```diff
@@ -1,12 +1,39 @@
 # Object Oriented CSS
 
+
+
 ## What is OOCSS
 
+OOCSS or 'Object Orientated CSS' is an approach to writing css that aims to encourage reusable, maintainable and predictable css. It has spawned a number of different approaches to standardise the concept most of which adhere to the core principles of OOCSS outline below.
+
+A CSS 'object' is a repeating visual pattern, that can be abstracted into an independent snippet of HTML, CSS, and possibly JavaScript. That object can then be reused throughout a project. The aim is to break down the css declaration into the smallest blocks possible in order to maximise their potential for reuse.
+
+An important component of OOCSS is the process of defining repeating visual features as separate modifying 'skins' that you can apply as distinctive classes to allow visual variety regardless of the structure of the HTML. In order to do this css declarations should not rely upon any assumptions about the semantics of the HTML. Instead classes are used to define 'objects' and the components therein. The layout and display of particular object or components can then be modified by additional classes or 'skins'.
+
+
+The two guiding principles of OOCSS could be defined as follows:
+
+- Separation of structure from skin: Keep the visuality separate, so you can reuse the visual classes
+
+- Separation  of container and content: Avoid the use of location-dependent styles and always aim to identify containers and components that can be reused.
+
+
+In this regard it privileges lack of repetition in the stylesheet over increases to the size of the HTML through the addition of extra classes.
+The hope is that in many cases the reduced css file size will out-weight the potential bloating of the HTML.
```

Comment:
Might be worth being a little stronger here - it's more than a hope.

In fact I'd recommend deleting this sentence. If you look at our html, you can't characterise it as "bloated". The only circumstance where I can see this actually being worthy of mention is if you had an incredibly high traffic site with multiple different page views where saving a few bytes on every page (through being able to cache all the css, and having smaller page weights due to fewer classes) could outweigh the maintainability benefits of OOCSS. So … like the traffic of Facebook, the variability of pages that we have on our site, and not using any form of edge caching.

"Reduced CSS size vs HTML bloating" isn't a thing :)